sycl/include/CL/sycl/detail/kernel_impl.hpp

```diff
@@ -33,7 +33,16 @@ class kernel_impl {
               std::shared_ptr<program_impl> ProgramImpl,
               bool IsCreatedFromSource)
       : ClKernel(ClKernel), Context(SyclContext), ProgramImpl(ProgramImpl),
-        IsCreatedFromSource(IsCreatedFromSource) {}
+        IsCreatedFromSource(IsCreatedFromSource) {
+    cl_context Context = nullptr;
+    CHECK_OCL_CODE(clGetKernelInfo(ClKernel, CL_KERNEL_CONTEXT, sizeof(Context),
```

Comment:
Can you use get_info<>() method for current object to get context? it should shorten the code and avoid direct OpenCL code

Reply:
No, I'm getting context of passed **OpenCL kernel object**, not current **SYCL kernel object**. I can't use SYCL `get_info<>` method for pure OpenCL objects.
I can use helper `get_kernel_info_cl` defined in detail/kernel_info.hpp:39 with `cl_kernel` parameter to get some information about `cl_kernel` object if it looks better for you.

Reply:
No, I'm getting context of passed OpenCL kernel object, not current SYCL kernel object. I can't use SYCL get_info<> method for pure OpenCL objects.
I can use helper get_kernel_info_cl defined in detail/kernel_info.hpp:39 with cl_kernel parameter to get some information about cl_kernel object if it looks better for you.

Reply:
it looks better for me.

Reply:
Oh, you are right, there is no specialization for `cl_context` return.
I'm not sure that I want to add it in the current patch.

Reply:
I don't see any value in moving this code to the helper at the moment. It's the only place where we query context for a kernel. I'm okay to commit as is.
@vladimirlaz are you okay?